packages/deploy-tooling/src/base/deploy.ts

```diff
@@ -281,6 +326,47 @@ async function getAbapService(config: AbapDeployConfig, logger: Logger): Promise
     return service;
 }
 
+/**
+ * Creates a new transport request using adt service.
+ *
+ * @param config - deployment configuration
+ * @param logger - reference to the logger instance
+ * @param provider - instance of the axios-extension abap service provider (it is created if not passed in)
+ * @returns transportRequest
+ */
+export async function createTransportRequest(
```

Comment:
This method can be consumed in Fiori deployment generator to replace the existing functionality around generating the TR value.
FYI @slin-sap